ADR/propagate-pull-secrets.md

```diff
@@ -0,0 +1,50 @@
+# 0. Support deploying images from Private Quay.io repositories
+
+Date: 2023-03-16
+
+## Status
+
+Provisional
+
+## Context
+
+* As a user, I would expect StoneSoup to generate a private Quay.io image repository if my source code repository is  private.
+* As a user, I would expect StoneSoup to gracefully deploy images that were pushed to private Quay.io repositories in every relevant environment.
```

Comment:
The expectation is that all Environments configured in a workspace should be able to pull images for components even if those images are private repositories?

Reply:
> all Environments configured 

This should only be for relevant environments, ex, those with SnapShotEnvironmentBindings.